File: ProgrammingChallenge/app/src/main/java/com/gabrielbmoro/programmingchallenge/models/Movie.kt
package com.gabrielbmoro.programmingchallenge.models

import com.google.gson.annotations.SerializedName

/**
 * This data class represents the Movie.
 * The annotations used was the Retrofit and Gson library.
 * @author Gabriel Moro
 * @since 2018-08-30
 */
data class Movie(
        @SerializedName("vote_count") val votes: Int,
        @SerializedName("id") val id: Int,
        @SerializedName("video") var isVideo: Boolean,
        @SerializedName("vote_average") var votesAverage: Float,
        @SerializedName("title") var title: String,
        @SerializedName("popularity") var popularity: Float,
        @SerializedName("poster_path") var posterPath: String,
        @SerializedName("original_language") var originalLanguage: String,
        @SerializedName("original_title") var originalTitle: String,
        @SerializedName("genre_ids") var genreIds: List<Int>? = null,
        @SerializedName("backdrop_path") var backdropPath: String,
        @SerializedName("adult") var isAdult: Boolean,
        @SerializedName("overview") var overview: String,
        @SerializedName("release_date") var releaseDate: String,
        var isFavorite: Boolean = false
)

File: flutter-version/ProgrammingChallenge/movie_db_app/lib/domain/model/movie.dart
import 'package:flutter/material.dart';

class Movie {
  int id;
  num votesAverage;
  String title;
  bool isVideo;
  int votes;
  num popularity;
  String postPath;
  String originalLanguage;
  String originalTitle;
  String backdropPath;
  bool isAdult;
  String overview;
  String releaseDate;
  bool isFavorite = false;

  Movie({
    this.id,
    @required this.votesAverage,
    @required this.title,
    @required this.isVideo,
    @required this.votes,
    @required this.popularity,
    @required this.postPath,
    @required this.originalLanguage,
    @required this.originalTitle,
    @required this.backdropPath,
    @required this.isAdult,
    @required this.overview,
    @required this.releaseDate,
  });

  String imageAddress() {
    return "https://image.tmdb.org/t/p/w500$postPath";
  }

  @override
  String toString() {
    return "$title -> popularity $popularity, votes $votesAverage";
  }
}
